src/Symfony/Component/SignalRegistry/SignalRegistry.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Symfony\Component\SignalRegistry;
+
+final class SignalRegistry implements SignalRegistryInterface
+{
+    private $signals = [];
+
+    public function __construct(bool $asynchronousMode = true)
+    {
+        if ($asynchronousMode) {
+            pcntl_async_signals(true);
+        }
+    }
+
+    public function register(int $signal, callable $callback): void
+    {
+        $this->signals[$signal][] = $callback;
+        pcntl_signal($signal, [$this, 'handler']);
```

Comment:
@lyrixx Symfony\Component\Messenger\Worker and Monolog also use pcntl_signal, haven't you had any problems you've told about? Like some pcntl_signal was redefined by the next one?

Reply:
yes, this will could be problematic!

Reply:
Php has a function called `pcntl_signal_get_handler` that lets to get previous handler. But there are few problems. 
- We can get previous handler, but some code can redefine this handler after. 
- The second problem is what's the right sequence for execution all handlers. In one of the handlers may be `exit`. In fact in one of the handlers likely will be `exit` otherwise the script won't be stopped. Probably we could add `exit` into "SignalRegistry->handler()", but
- the function becomes hard to cover with tests
- what if the developer don't want to exit? What if they handled SIGUSR1?

Reply:
@lyrixx, could you, please, look at the merge request again and say what do you think?

Reply:
Nice :+1: